app/helpers/vaccine_query_helper.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+# Helper methods for vaccine-related queries
+module VaccineQueryHelper
+  # Validates the request params for the index action
+  def validate_table_query(params)
+    permitted_params = params.permit(:entries, :page, :search, :order, :direction)
+
+    patient_id = params.require(:patient_id)&.to_i
+    search_text = permitted_params[:search]
+    sort_order = permitted_params[:order]
+    sort_direction = permitted_params[:direction]
+    # Assume default values if pagination data is not explicitly specified
+    entries = permitted_params[:entries]&.to_i || 10
+    page = permitted_params[:page]&.to_i || 0
+
+    if sort_direction.present? && !%w[asc desc].include?(sort_direction)
+      error_message = "Unable to sort column in specified direction in request: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    if sort_order.present? && !%w[id group_name product_name administration_date dose_number notes].include?(sort_order)
+      error_message = "Unable to sort by specified column in request: '#{sort_order}'"
+      raise StandardError, error_message
+    end
+
+    # Either both the sort order and direction are provided, or neither is provided
+    if (sort_direction.blank? && sort_order.present?) || (sort_direction.present? && sort_order.blank?)
+      error_message = 'Must have both a sort column and direction specified or neither specified. Requested column to sort: '\
+                      "'#{sort_order}'', with specified direction: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    if entries.negative? || page.negative?
+      error_message = "Invalid pagination options. Number of entries: #{entries}. Page: #{page}"
+      raise StandardError, error_message
+    end
+
+    {
+      patient_id: patient_id,
+      search_text: search_text,
+      sort_order: sort_order,
+      sort_direction: sort_direction,
+      entries: entries,
+      page: page
+    }
+  end
+
+  # Queries vaccines by ID or type based on search text.
+  def search(vaccines, search)
```

Comment:
is it too expensive to search by notes?

Reply:
I had asked about what fields users may want to search by for this and the PO team agreed this functionality likely won't even be used. Given that, we felt it was okay to limit it to a few fields at the moment.